src/ol/interaction/Link.js

```diff
@@ -342,6 +368,46 @@ class Link extends Interaction {
     }
   }
 
+  /**
+   * Register a listener for a URL search parameter.  The callback will be called with a new value
+   * when the corresponding search parameter changes due to history events (e.g. browser navigation).
+   *
+   * @param {string} key The URL search parameter.
+   * @param {Callback} callback The function to call when the search parameter changes.
+   * @return {string|null} The initial value of the search parameter (or null if absent from the URL).
+   * @api
+   */
+  track(key, callback) {
+    this.trackedCallbacks_[key] = callback;
+    const url = new URL(window.location.href);
+    const params = url.searchParams;
+    const value = params.get(key);
+    this.trackedValues_[key] = value;
+    return value;
+  }
+
+  /**
+   * Update the URL with a new search parameter value.  If the value is null, it will be
+   * deleted from the search parameters.
+   *
+   * @param {string} key The URL search parameter.
+   * @param {string} [value] The updated value (or null to remove it from the URL).
```

Comment:
This allows `undefined` and omitting the value parameter.
Should this be `{string|null} value`?

Reply:
Good catch.  Updated in the latest commit.